source/common/common/packed_struct.h

```diff
@@ -131,9 +129,7 @@ template <class T, uint8_t max_size, class ElementName> class PackedStruct {
     // If we're at capacity and we don't have a slot for element_idx, increase capacity by 1.
     if (!has(element_idx) && current_size == capacity()) {
       std::unique_ptr<T[]> tmp(new T[++indices_[max_size]]);
-      for (size_t idx = 0; idx < current_size; ++idx) {
-        swap(tmp.get()[idx], data_.get()[idx]);
-      }
+      std::move(data_.get(), std::next(data_.get(), current_size), tmp.get());
```

Comment:
this is looking more like std::vector::insert :) Any reason not to use that?

Reply:
Mainly because sizeof(std::vector) is 24. That's additional 16 bytes.

Reply:
that's quite a good reason :)